core/src/main/scala/org/apache/spark/SparkHadoopWriter.scala

```diff
@@ -105,10 +107,20 @@ class SparkHadoopWriter(@transient jobConf: JobConf)
   def commit() {
     val taCtxt = getTaskContext()
     val cmtr = getOutputCommitter()
+    val dagSchedulerActor =
+      AkkaUtils.makeDriverRef("DAGScheduler", SparkEnv.get.conf, SparkEnv.get.actorSystem)
+    val askTimeout = AkkaUtils.askTimeout(SparkEnv.get.conf)
     if (cmtr.needsTaskCommit(taCtxt)) {
       try {
-        cmtr.commitTask(taCtxt)
-        logInfo (taID + ": Committed")
+        val canCommit: Boolean = AkkaUtils.askWithReply(
+          AskPermissionToCommitOutput(jobID, splitID, attemptID), dagSchedulerActor, askTimeout)
```

Comment:
Whether or not such an extra layer of complexity and coordination is necessary... it would be good before we merge this patch (or some version of it) to run a benchmarking suite against it. Even though this is only affecting hadoop-writing result tasks, it's on a critical path and we need a good understanding of the performance implications.